### ISSUE TITLE
Bug with roberta model using retromae_pretrain module

### ISSUE THREAD

**Issue by dinhvietcuong** [FlagOpen/FlagEmbedding#513]
Hi team,

When I ran retromae_pretrain module using the pretrained model "distilroberta-base", I ran into this error message: 

===============
File "/root/.cache/pypoetry/virtualenvs/nir-lKa0R1gD-py3.10/lib/python3.10/site-packages/FlagEmbedding/baai_general_embedding/retromae_pretrain/modeling.py", line 56, in forward
decoder_position_ids = self.lm.bert.embeddings.position_ids[:, :decoder_input_ids.size(1)]

File "/root/.cache/pypoetry/virtualenvs/nir-lKa0R1gD-py3.10/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1269, in __getattr__
raise AttributeError("'{}' object has no attribute '{}'".format(
AttributeError: 'RobertaForMaskedLM' object has no attribute 'bert'
===============

I think the issue is because of the line 56-57 in modeling.py: 
https://github.com/FlagOpen/FlagEmbedding/blob/master/FlagEmbedding/baai_general_embedding/retromae_pretrain/modeling.py#L56-L57

in which `self.lm.bert.embeddings` should be replaced by `self.decoder_embeddings`. 

Can you please check?

**Comment by staoxiao** [comment]
Yes. Another approach is to employ code in lines [60-61](https://github.com/FlagOpen/FlagEmbedding/blob/master/FlagEmbedding/baai_general_embedding/retromae_pretrain/modeling.py#L60-L61) instead of lines 56-58,.

**Comment by dinhvietcuong** [comment]
Thanks @staoxiao for your quick response and the confirmation.
I have another related question:

I saw quite a few warnings when running retromae_pretrain module using the input model `distilroberta-base`:

```transformers/data/data_collator.py:984: UserWarning: DataCollatorForWholeWordMask is only suitable for BertTokenizer-like tokenizers. Please refer to the documentation for more information.```

Would that be a serious problem that affects the accuracy of the resulting model? And if yes, is there a way to fix or get around it?

thank you!

**Comment by dinhvietcuong** [comment]
Clear. Thanks for the explanation!
